### PR TITLE
Add tests for `fileio`

### PR DESCRIPTION
**Pull request by alex-zenml**
## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Describe changes
- Added tests for our `fileio` functionality.
- Fixed places where `get_grandparent` and `get_parent` were being called inside the codebase, since I fixed this in `fileio` to what makes sense.

### CODE REVIEW

**Review by AlexejPenner — APPROVED** [review]
Thats a lot of tests, well done :fire:

**Comment by alex-zenml** [comment]
@bcdurak and/or @htahir1 would appreciate your eyes on this. intersects with `tfx` for obv reasons and I want to be sure I didn't approach this in the wrong way.

File: tests/unit/io/test_fileio.py [hunk 1]
```diff
@@ -89,14 +94,16 @@ def test_find_files_when_file_absent(tmp_path):
 
 @pytest.mark.parametrize("filesystem", REMOTE_FS_PREFIX)
 def test_is_remote_when_using_remote_prefix(filesystem):
-    """is_remote returns True when path starts with one of the TFX remote file prefixes"""
+    """is_remote returns True when path starts with one of
+    the TFX remote file prefixes"""
```
Comment (bcdurak) [inline tests/unit/io/test_fileio.py:98]: The docstring might be a bit misleading here, as the check is based on the prefixes which reside in our `zenml.constants`

**Review by bcdurak — APPROVED** [review]
The new additions to our tests seem great overall!

File: tests/unit/io/test_fileio.py (hunk 1 above)
Comment (alex-zenml) [reply tests/unit/io/test_fileio.py]: Fixed.

**Review by htahir1 — APPROVED** [review]
LGTM! Awesome job on these tests

**Pull request merged**